### ISSUE TITLE
Add hash template functions

### ISSUE THREAD

**Issue by kakawait** [spf13/hugo#1910]
It will be nice to have ability to hash content on template.

Two possibilities:

1. A generic hash function `{{ hash <CONTENT> <ALGORITHM> }}`
2. Common algorithm hash function pipeline `{{ .Site.Params.my_var | md5 }}`

Good candidates should be:

- `md5` 
- `sha1`
- `sha2`

--- 

*Use case*: generate *Gravatar* url from `email`

```
<meta property="og:image" content="http://www.gravatar.com/avatar/{{ .Site.Params.gravatar.email | md5 }}">
<meta property="og:image" content="http://www.gravatar.com/avatar/{{ hash .Site.Params.gravatar.email "md5" }}">
```

**Comment by bep** [comment]
A fairly narrow use case ... Not sure if it is worth the cost as there are simple workarounds.

**Comment by digitalcraftsman** [comment]
Hello @kakawait,

I added some of the template funcs that you requested, currently `md5` and `sha1`. Could you be a bit more specific which functions you exactly want to see implemented?

[SHA-2](https://en.wikipedia.org/wiki/SHA-2) and [SHA-3](https://en.wikipedia.org/wiki/SHA-3) are sets of hashing-functions. I don't it makes sence to add all of them.

**Comment by bep** [comment]
I think it is more a question of "what do we want" to add. md5 and sha1 should cover what Hugo is concerned about.

**Comment by digitalcraftsman** [comment]
That's exactly my thought.

**Comment by kakawait** [comment]
Yes `md5` and `sha1` should cover 99% of use cases. Otherwise if hash function is too edge case to be core on Hugo no prob. I will simply store md5 directly inside config

**Comment by digitalcraftsman** [comment]
Merged as 94c3825e5bdda9c57b1e8782caa3c407cfad711e
